app/lib/replay/ReplayProblems.ts

```diff
@@ -0,0 +1,210 @@
+  } catch (error) {
+    console.error('Error fetching problems', error);
+    toast.error('Failed to fetch problems');
+    return [];
+  }
+}
+
+export async function replayGetProblem(problemId: string): Promise<BoltProblem | null> {
+  try {
+    const rv = await sendCommandDedicatedClient({
+      method: 'Recording.globalExperimentalCommand',
+      params: {
+        name: 'fetchBoltProblem',
+        params: { problemId },
+      },
+    });
+    console.log('FetchProblemRval', rv);
+    const problem = (rv as { rval: { problem: BoltProblem } }).rval.problem;
+    if ('prompt' in problem) {
+      // 2/11/2025: Update obsolete data format for older problems.
+      problem.repositoryContents = (problem as any).prompt.content;
+      delete problem.prompt;
+    }
+    return problem;
+  } catch (error) {
+    console.error('Error fetching problem', error);
+    toast.error('Failed to fetch problem');
+  }
+  return null;
+}
+
+export async function replaySubmitProblem(problem: BoltProblemInput): Promise<string | null> {
+  try {
+    const rv = await sendCommandDedicatedClient({
+      method: 'Recording.globalExperimentalCommand',
+      params: {
+        name: 'submitBoltProblem',
+        params: { problem },
+      },
+    });
+    console.log('SubmitProblemRval', rv);
+    return (rv as any).rval.problemId;
+  } catch (error) {
+    console.error('Error submitting problem', error);
+    toast.error('Failed to submit problem');
+    return null;
+  }
+}
+
+export async function replayUpdateProblem(problemId: string, problem: BoltProblemInput | undefined) {
+  try {
+    if (!getNutIsAdmin()) {
+      toast.error('Admin user required');
+      return;
+    }
+
+    const loginKey = Cookies.get(nutLoginKeyCookieName);
+    await sendCommandDedicatedClient({
+      method: 'Recording.globalExperimentalCommand',
+      params: {
+        name: 'updateBoltProblem',
+        params: { problemId, problem, loginKey },
+      },
+    });
+  } catch (error) {
+    console.error('Error updating problem', error);
+    toast.error('Failed to update problem');
+  }
+}
+
+const nutLoginKeyCookieName = 'nutLoginKey';
+const nutIsAdminCookieName = 'nutIsAdmin';
+
+export function getNutLoginKey(): string | undefined {
+  const cookieValue = Cookies.get(nutLoginKeyCookieName);
+  return cookieValue?.length ? cookieValue : undefined;
+}
+
+export async function saveNutLoginKey(key: string) {
+  const {
+    rval: { userInfo },
+  } = (await sendCommandDedicatedClient({
+    method: 'Recording.globalExperimentalCommand',
+    params: {
+      name: 'getUserInfo',
+      params: { loginKey: key },
+    },
+  })) as { rval: { userInfo: UserInfo } };
+  console.log('UserInfo', userInfo);
+
+  Cookies.set(nutLoginKeyCookieName, key);
+  Cookies.set(nutIsAdminCookieName, userInfo.admin ? 'true' : 'false');
+}
+
+export function getNutIsAdmin(): boolean {
+  return Cookies.get(nutIsAdminCookieName) === 'true';
+}
+
+interface UserInfo {
```

Comment:
Why did the `UserInfo` interface get moved? It looks like this is the legacy code?
I cannot tell what changed in this file.

For review + diff purposes, we should keep legacy filenames as-is, and put new code into new files, so we can see proper diffs.

Reply:
That's fair. I will go back and try and optimize for the smallest PR.